File: app/src/main/java/com/lear7/showcase/fragment/UiFragment.java
package com.lear7.showcase.fragment;

import android.view.View;
import android.widget.Button;

import com.lear7.showcase.R;
import com.lear7.showcase.constants.Routers;

import butterknife.BindView;
import butterknife.OnClick;

public class UiFragment extends BaseFragment {

    @BindView(R.id.btn_constraint)
    Button btnGoToB;

    @BindView(R.id.btn_material)
    Button btnMaterialSimple;

    @BindView(R.id.btn_material_alipay)
    Button btnMaterial;

    @BindView(R.id.btn_listview)
    Button btnListView;

    @OnClick({R.id.btn_constraint,
            R.id.btn_material,
            R.id.btn_material_alipay,
            R.id.btn_listview,
    })
    public void onClick(View view) {
        if (view == btnGoToB) {
            goTo(Routers.Act_Constaint);
        } else if (view == btnMaterial) {
            goTo(Routers.Act_MaterialAlipay);
        } else if (view == btnListView) {
            goTo(Routers.Act_ListView);
        } else if (view == btnMaterialSimple) {
            goTo(Routers.Act_Material);
        }
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_ui;
    }
}

File: app/src/main/java/com/lear7/showcase/fragment/UiFragment.kt
package com.lear7.showcase.fragment

import com.lear7.showcase.R
import com.lear7.showcase.constants.Routers
import kotlinx.android.synthetic.main.fragment_ui.*

class UiFragment : BaseFragment() {
    override fun getLayoutId(): Int {
        return R.layout.fragment_ui
    }

    override fun initView() {
        super.initView()
        btn_constraint.setOnClickListener { goTo(Routers.Act_Constaint) }
        btn_material.setOnClickListener { goTo(Routers.Act_Material) }
        btn_material_alipay.setOnClickListener { goTo(Routers.Act_MaterialAlipay) }
        btn_listview.setOnClickListener { goTo(Routers.Act_ListView) }
        btn_customview.setOnClickListener{goTo(Routers.Act_Printer)}
    }
}